lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -108,3 +109,11 @@ def test_dataset_in_given_path(self, query_get_as, monkeypatch):
         )
         response = query_get_as("fio_2", "1-default/default.csv", HTTPStatus.OK)
         assert response.status_code == HTTPStatus.OK
+
+    def test_get_result_tarball(self, query_get_as, monkeypatch):
+        monkeypatch.setattr(FileTree, "find_dataset", self.mock_find_dataset)
+        monkeypatch.setattr(
+            werkzeug.utils, "send_file", lambda *args, **kwargs: {"status": "OK"}
+        )
+        response = query_get_as("fio_2", "", HTTPStatus.OK)
+        assert response.status_code == HTTPStatus.OK
```

Comment:
I apparently didn't think to point this out in the original PR, but `test_dataset_in_given_path` has the same problem: we're checking for `OK` status but never validating that the `send_file` mock was called. This is a disadvantage of the cute lambda implementation. Using a method or function would allow capturing some state that could be tested to verify that the call was made, and even to verify that the expected parameters were passed.

This probably isn't a big deal for test coverage, but it eliminates one way we might be able to get a false positive on the test case. (That is, something generates the `OK` return status without actually making the call to return the byte stream.)